### PR TITLE
Tests create new

### PR DESCRIPTION
**Pull request by SPlanzer**
Fixes: #
address - https://github.com/linz/ds-group-epics/issues/133
The main purpose of this is to test adding a new place feature/name and editing its properties. 

This also introduces an entire approach to testing the plugin features and managing test data

Happy to talk anyone through this in person

### Change Description:

...

### Notes for Testing:
Let CI run
...

#### Source Code Documentation Tasks:
- [ ] README updated (where applicable)
- [ ] CHANGELOG updated

#### User Documentation Tasks:
- [ ] Confluence user guide updated (where applicable)

#### Testing Tasks:
- [x] Added tests that fail without this change
- [x] All tests are passing in development environment
- [x] Reviewers assigned

#### Pull Request Management:
- [ ] Linked to sub-task
- [ ] Linked to the epic

### CODE REVIEW

**Comment by SPlanzer** [comment]
FYI @strk as talked about - here is a draft PR

I need to review why tests are failing in CI and then I will seek review

**Comment by SPlanzer** [comment]
@billgeo @strk This is at the point that it is ready for review. 

Apologies this is very large. I am happy to talk in our next meeting how to best manage review on this project

**Comment by billgeo** [comment]
Thanks @SPlanzer. I see all the CI is passing. 

Maybe I can review the test coverage and @strk can review the approach/framework for running the tests?

Can you confirm what tests in https://confluence.linz.govt.nz/display/MNAI/NZGB+Gazetter+Plugin+Tests are delivered as part of this PR?

**Comment by SPlanzer** [comment]
Can you confirm what tests in https://confluence.linz.govt.nz/display/MNAI/NZGB+Gazetter+Plugin+Tests are delivered as part of this PR?

> This covers - Written "Test Plan #1: Create a new entry in the Gazetteer, populating all fields"

Coverage is intentionally not 100%

File: src/NZGBplugin/Ui_NewFeatureDialog.py [hunk 1]
```diff
@@ -1,21 +1,8 @@
...
-#  This file is released under the MIT licence. See the LICENCE file found
-#  in the top-level directory of this distribution for more information.
-#
-################################################################################
-
-
 # Form implementation generated from reading ui file 'Ui_NewFeatureDialog.ui'
 #
-# Created by: PyQt5 UI code generator 5.9.2
+# Created by: PyQt5 UI code generator 5.10.1
```
Comment (strk) [inline src/NZGBplugin/Ui_NewFeatureDialog.py:5]: Just wondering.. does a generated file belong to the repository at all ? (something for later, probably)

**Pull request merged**